### ISSUE TITLE
Templated classes

### ISSUE THREAD

**Issue by noizex** [baszalmstra/header-parser#3]
Hi, I really like this small library - it allowed me to prepare bindings for my scripting language and automate this process with just a bit of meta annotations on my C++ classes. One thing I couldn't get to work is annotating templated classes - I mainly use it as templated base class to some classes that are registered to scripting. Would it be possible to parse this information too using header-parser?

**Comment by baszalmstra** [comment]
I think template classes could be added, the only difficulty is with template default arguments I guess. Do you have an example of your specific use case so that I can work out a best approach?

**Comment by baszalmstra** [comment]
I went ahead and implemented a simple solution in PR #4 . Would that be good enough for you? If so, I'll merge that.

**Comment by noizex** [comment]
Wow, that was fast :) Thanks! Yeah I needed a way to parse templated classes - somehow with a class like this:
```
template<class T>
class myClass {}
```
it would parse only `template` token and skip the class. If your change makes it possible to get to that templated class, and having template params info - that's perfect!

One last thing that I was wondering is - parsing the inheritance information - when registring classes for scripting in my case, some of them have base class that should be registered together with the derived class (so basically for every derived class, scripting language I use requires me to register both: derived and parent class). If I could establish a link between these classes it would be trivial to register them together - right now I need to provide some meta info on top of that rather than infer it from inheritance. 

But this would be cherry on top - the template parsing will work great. I tried to look into the parser but couldn't figure out how to add this easily.  I will try this today and give some feedback.

**Comment by noizex** [comment]
Wow, that was fast :) Thanks! Yeah I needed a way to parse templated classes - somehow with a class like this:
```
template<class T>
class myClass {}
```
it would parse only template token and skip the class. If your change makes it possible to get to that templated class, and having template params info - that's perfect!

I tried to look into the parser but couldn't figure out how to add this easily. I will try this today and give some feedback.

**Comment by noizex** [comment]
I checked and it works great and as expected, thanks! While we're discussing improvements - is there any particular reason why `-c` switch is limited to single entry? I found it useful to have more macros that annotate the class - I could use some meta info within the same macro but it looks much more transpartent when using macro name. I changed it locally as it was switching from single to multiple arguments and vector search rather than string comparison - maybe it would be worth to remove this limitation?
